project/src/store/reducer.ts

```diff
@@ -0,0 +1,20 @@
+import { createReducer } from '@reduxjs/toolkit';
+import {changeGenre, getFilmListByGenre} from './action';
+import { FilmMockTypes } from '../types/films-mock-type';
+import { DEFAULT_GENRE } from '../const';
+
+const initialState = {
+  genre: DEFAULT_GENRE,
+  films: [] as FilmMockTypes,
+};
+
+export const reducer = createReducer(initialState, (builder) => {
+  builder.addCase(changeGenre, (state, action) => {
+    const {genre} = action.payload;
+    state.genre = genre;
+  })
+    .addCase(getFilmListByGenre, (state, action) => {
+      const {films} = action.payload;
+      state.films = films;
```

Comment:
экшены нужны чтобы данные в сторе менять, поэтмоу из глаголов обыно использую change или set, поменяй название хотя бы на 
```suggestion
    .addCase(setFilmListByGenre, (state, action) => {
      const {films} = action.payload;
      state.films = films;
```

Reply:
но при этом если хранишь фильмы по конкретному жанру, храни тогда и общий список, потому что это сейчас ты можешь его из моков взять, а потом придётся запросы на сервер делать, а вообще я бы на месте в компоненте фильтровал по жанру